cmd/metal-api/internal/service/firewall-service.go

```diff
@@ -215,6 +231,42 @@ func (r *firewallResource) allocateFirewall(request *restful.Request, response *
 	r.send(request, response, http.StatusOK, resp)
 }
 
+func (r firewallResource) setVPNConfigInSpec(allocationSpec *machineAllocationSpec) error {
+	if r.headscaleClient == nil {
+		return nil
+	}
+
+	ctx := context.Background()
+
+	// Get project VPN namespace
+	projectID := allocationSpec.ProjectID
+	p, err := r.mdc.Project().Get(ctx, &mdmv1.ProjectGetRequest{Id: projectID})
+	if err != nil {
+		return err
+	}
+	if p.GetProject() == nil {
+		return fmt.Errorf("Project with ID %s wasn't found", projectID)
+	}
+
+	// Try to create namespace in Headscale DB
+	if err = r.headscaleClient.CreateNamespace(p.Project.Name); err != nil {
```

Comment:
Decided to use the project ID for the namespace name, this way you also do not need to fetch the project from the masterdata api.